Objects/longobject.c

```diff
@@ -262,6 +262,35 @@ _PyLong_FromSTwoDigits(stwodigits x)
     return _PyLong_FromLarge(x);
 }
 
+int
+_PyLong_AssignValue(PyObject **target, long value)
```

Comment:
Could you make this `Py_ssize_t` rather than `long`?
`sizeof(long) != sizeof(void *)` which makes reasoning about that much harder.

Reply:
Should we change PyRangeIterator to use Py_ssize_t as well?

Reply:
Ultimately yes. But maybe not in this PR, since that isn't new code.